### DISCUSSION TITLE
What's the point of this call to `source`?

### DISCUSSION THREAD

**Discussion by ivanperez-keera** [space-ros/docker#109] [Q&A]
I just noticed that we are all over the place calling `bash -c 'source <path>'` before we execute ros-related commands. To me, this makes sense under the assumption that the adjustments made by those invocations do not persist across docker `RUN`s.

But then I noticed the same at the very end of a `RUN`:
https://github.com/space-ros/docker/blob/76cd412a7379dd5a5ec338b7b75659afbbf0c9a1/space_robots/Dockerfile#L78

What's the point of doing that?

If the effects of calling `source` persist across `RUN`s, then should these be simplified throughout the file? Note that just removing duplication is not enough of a reason to factorize those; there are circumstances under which you want to spell it out twice.

If the effects of calling `source` do not persist across `RUN`s, then we should just remove that last bit from the line above.

**Comment by ivanperez-keera** [comment]
I'm opening this as a question first because it's likely that I'm just misunderstanding something. Please do me an educate :smile:

**Comment by Bckempa** [comment]
This could be combined with https://github.com/space-ros/docker/issues/102 / https://github.com/space-ros/docker/discussions/108 into a dockerfile cleanup issue once https://github.com/space-ros/space-ros/discussions/114 is resolved

**Comment by EzraBrooks** [comment]
> If the effects of calling source do not persist across RUNs, then we should just remove that last bit from the line above.

This is the case. Each `RUN` directive is executed in a separate sub-shell for idempotence/reproducibility reasons.
